flink-applications.md

```diff
@@ -0,0 +1,203 @@
+---
+title: "What is Apache Flink?"
+---
+
+<hr/>
+<div class="row">
+  <div class="col-sm-12" style="background-color: #f8f8f8;">
+    <h2>
+      <a href="{{ site.baseurl }}/flink-architecture.html">Architecture</a> &nbsp;
+      <span class="glyphicon glyphicon-chevron-right"></span> &nbsp;
+      Applications &nbsp;
+      <span class="glyphicon glyphicon-chevron-right"></span> &nbsp;
+      <a href="{{ site.baseurl }}/flink-operations.html">Operations</a>
+    </h2>
+  </div>
+</div>
+<hr/>
+
+Apache Flink is a framework for stateful computations over unbounded and bounded data streams. Flink provides multiple APIs at different levels of abstraction and offers dedicated libraries for common use cases.
+
+Here, we present Flink's easy-to-use and expressive APIs and libraries.
+
+## Building Blocks for Streaming Applications
+
+The types of applications that can be built with and executed by a stream processing framework are defined by how well the framework controls *streams*, *state*, and *time*. In the following, we describe these building blocks for stream processing applications and explain Flink's approaches to handle them.
+
+### Streams
+
+Obviously, streams are a fundamental aspect of stream processing. However, streams can have different characteristics that affect how a stream can and should be processed. Flink is a versatile processing framework that can handle any kind of stream.
+
+* **Bounded** and **unbounded** streams: Streams can be unbounded or bounded, i.e., fixed-sized data sets. Flink has sophisticated features to process unbounded streams, but also dedicated operators to efficiently process bounded streams. 
+* **Real-time** and **recorded** streams: All data are generated as streams. There are two ways to process the data. Processing it in real-time as it is generated or persisting the stream to a storage system, e.g., a file system or object store, and processed it later. Flink applications can process recorded or real-time streams.
+
+### State
+
+Every non-trivial streaming application is stateful, i.e., only applications that apply transformations on individual events do not require state. Any application that runs basic business logic needs to remember events or intermediate results to access them at a later point in time, for example when the next event is received or after a specific time duration.
+
+<div class="row front-graphic">
+  <img src="{{ site.baseurl }}/img/function-state.png" width="350px" />
+</div>
+
+Application state is a first-class citizen in Flink. This is easy to see by looking at all the features that Flink provides in the context of state handling.
```

Comment:
The second sentence reads like we're trying to proof something.

Reply:
Not sure, does not sound like that to me.
`You can see that by looking at all the features that Flink provides in the context of state handling.` would sound nicer, though, in my opinion.

Reply:
+1 for Stephan's suggestion